File: python/7 kyu/93.py
# https://www.codewars.com/kata/62f96f01d67d0a0014f365cf/train/python

# Passed

def ball_test(s, r):
    while s > 0 and r:
        cracks = r[:s].count("x")
        r = r[s:]
        s -= cracks + 1

        if not r:
            # road has been completed
            return True

    return False

output = ball_test(24, 'xxxxxxxxxx_____x___xx__xx____________x__________x_')
print(output)

File: nodejs/7 kyu/93.js
// https://www.codewars.com/kata/54b81566cd7f51408300022d/train/javascript

// Passed

function wordSearch(query, seq) {
  const partialWords = seq.filter((word) =>
    word.toLowerCase().includes(query.toLowerCase())
  );
  const result = partialWords.length > 0 ? partialWords : ["Empty"];
  return result;
}

const output = wordSearch("ab", ["za", "ab", "abc", "zab", "zbc"]);
console.log(output);